src/Workspaces/Core/Portable/FindSymbols/DeclaredSymbolInfo.cs

```diff
@@ -90,12 +96,12 @@ internal struct DeclaredSymbolInfo
             TextSpan span,
             ImmutableArray<string> inheritanceNames,
             int parameterCount = 0, int typeParameterCount = 0)
-            : this()
         {
-            Name = name;
-            NameSuffix = nameSuffix;
-            ContainerDisplayName = containerDisplayName;
-            FullyQualifiedContainerName = fullyQualifiedContainerName;
+            var stringTable = s_projectStringTable.GetOrCreateValue(project);
```

Comment:
❓ Is this called in concurrent code? If so it could introduce a performance problem related to the use of `lock` in ConditionalWeakTable.

💭 You may see good benefits from using an opportunistic wait-free global cache, being nothing more than a `WeakReference<StringTable>` that you assign a value to should you ever obtain a null target from it.